internal/dag/builder_test.go

```diff
@@ -1159,6 +1159,73 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 				},
 			),
 		},
+		"insert basic single route, single hostname, gateway with TLS, HTTP protocol is ignored": {
+			gateway: &gatewayapi_v1alpha1.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "contour",
+					Namespace: "projectcontour",
+				},
+				Spec: gatewayapi_v1alpha1.GatewaySpec{
+					Listeners: []gatewayapi_v1alpha1.Listener{{
+						Port:     443,
```

Comment:
just noticed this, we're asking for a HTTP listener on port 443 and we end up getting one on port 80

Reply:
probably b/c of this https://github.com/projectcontour/contour/blob/9e7b4e6aad08be068e5566b757f8f1a76a6a80ba/internal/dag/gatewayapi_processor.go#L403-L406 but it is a bit counterintuitive

maybe so it is not confusing just change the port in the test to 80 for now and we can figure out what to do with the whole mismatched/one port thing separately

Reply:
The ports aren't used right now from Gateway listeners.

Reply:
#3616

Reply:
ah gotcha, that makes sense

this test may fail when we resolve the issue you created, but sounds good for now, HTTP protocol always gets you port 80